java/src/org/openqa/selenium/grid/commands/Hub.java

```diff
@@ -182,6 +182,12 @@ protected Handlers createHandlers(Config config) {
 
     Routable routerWithSpecChecks = router.with(networkOptions.getSpecComplianceChecks());
 
+    Boolean disableUi = new RouterOptions(config).disableUi();
+    if (disableUi) {
+      LOG.info("UI disabled");
```

Comment:
Same concern here.

Reply:
Thanks @pujagani for the feedback, I made the changes based on this.